src/base.jl

```diff
@@ -192,8 +193,59 @@ function ConcreteSystem(sys::System)
     return concrete_sys
 end
 
+
+"""Primary System constructor. Funnel point for all other outer constructors."""
+function System(buses::Vector{Bus},
+                generators::Vector{<:Generator},
+                loads::Vector{<:ElectricLoad},
+                branches::Union{Nothing, Vector{<:Branch}},
+                storage::Union{Nothing, Vector{<:Storage}},
+                basepower::Float64,
+                forecasts::Union{Nothing, SystemForecasts},
+                services::Union{Nothing, Vector{ <: Service}},
+                annex::Union{Nothing,Dict}; kwargs...)
+
+    
+    _sys = _System(buses, generators, loads, branches, storage, basepower, forecasts, services, annex; kwargs...)
+    return System(_sys)
+end
+
```

Comment:
I like your choice of constructors.  But didn't @jd-lara say that he wanted explicit constructors that dealt with optional Branches and Loads?

Reply:
08abf44e91b479f54c265022309060ef23582340 resolves